torchgeo/datasets/geo.py

```diff
@@ -46,20 +48,40 @@ class GeoDataset(Dataset[Dict[str, Any]], abc.ABC):
     * :term:`coordinate reference system (CRS)`
     * resolution
 
-    These kind of datasets are special because they can be combined. For example:
+    :class:`GeoDataset` is a special class of datasets. Unlike :class:`VisionDataset`,
+    the presence of geospatial information allows two or more datasets to be combined
+    based on latitude/longitude. This allows users to do things like:
 
-    * Combine Landsat8 and CDL to train a model for crop classification
-    * Combine NAIP and Chesapeake to train a model for land cover mapping
+    * Combine image and target labels and sample from both simultaneously
+      (e.g. Landsat and CDL)
+    * Combine datasets for multiple image sources for multimodal learning or data fusion
+      (e.g. Landsat and Sentinel)
 
-    This isn't true for :class:`VisionDataset`, where the lack of geospatial information
-    prohibits swapping image sources or target labels.
-    """
+    These combinations require that all queries are present in *both* datasets,
+    and can be combined using an :class:`IntersectionDataset`:
+
+    .. code-block:: python
+
+       dataset = landsat & cdl
+
+    Users may also want to:
 
-    #: :term:`coordinate reference system (CRS)` for the dataset.
-    crs: CRS
+    * Combine datasets for multiple image sources and treat them as equivalent
+      (e.g. Landsat 7 and Landsat 8)
```

Comment:
While working on this, I realized there are actually 4 different use cases. The multiple image sources can be interpreted two different ways:

1. Completely different image sources (e.g. Landsat and Sentinel)
2. Approximately equivalent image sources (e.g. Landsat 7 and 8)

The former requires IntersectionDataset while the latter requires UnionDataset.

Reply:
Right, for the second point here you could also imagine disparate datasets that all have RGB